sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridClient.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Messaging.EventGrid.Models;
+
+namespace Azure.Messaging.EventGrid.Customization
+{
+    public class EventGridClient
+    {
+        private readonly ServiceRestClient _serviceRestClient;
+        private readonly ClientDiagnostics _clientDiagnostics;
+        private readonly string _hostName;
+        private const string SasKeyName = "aeg-sas-key";
```

Comment:
Perhaps we can move these to a Constants class and have them be internal. We can then access them for tests once we add internals visible to for our test project:
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/src/Properties/AssemblyInfo.cs#L7